### PR TITLE
docs: Quickstart guide on getting started with novu in python

### PR DESCRIPTION
**Pull request by ankur0904**
### What change does this PR introduce?
This PR introduce the Getting started with Python Quickstart. 
As this quickstart guide is not available from the [quickstart section](https://docs.novu.co/overview/quickstart/general-quickstart/) 

### Why was this change needed?
Closes [#39](https://github.com/novuhq/novu-python/issues/39) 

### Other information (Screenshots)

[a.webm](https://github.com/novuhq/novu/assets/98346896/bc708d10-9e1e-4278-a3e5-aa94a10c58a3)

### CODE REVIEW

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+---
+sidebar_position: 10
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:2]: `sidebar_position: 10` already belongs to Ruby quickstart. Please change this

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+sidebar_label: Get started with Python
+---
+
+# Python Quickstart
+
+Welcome to the Python Quickstart guide for Novu, a powerful notification service that enables you to send multi-channel (SMS, Email, Chat, Push) notifications from your Python applications. In this Quickstart, you'll learn how to seamlessly integrate Novu into your Python project and perform various essential tasks. Let's get started!
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:8]: Welcome to Python Quickstart guide for Novu. In this guide, you'll learn how to use Novu in a Python application. Novu is a powerful notification service that enables you to send multi-channel (SMS, Email, Chat, Push) notifications. Let's see how you can seamlessly integrate Novu into your Python project!

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+## Prerequisites
+
+Before diving into the Quickstart, make sure you have the following:
+
+- Python3 installed on your development machine.
+- A Novu account. If you don't have one, sign up for free at [web.novu.co](https://web.novu.co)
+
+You can also [view the completed code](https://web.novu.co) of this quick start in a GitHub repo.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:17]: Not sure why this link is here. Have you made any demo app for this?

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+### Install and Set Up Novu in your Python Project
+
+First, you must install the Novu package in your Python project. You can install the Novu package in your project in two ways. Open your terminal and run the following command:
+
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:22]: First, you must install the Novu package in your Python project. From your terminal, you can install the Novu package in your project by running either of the following two commands:

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+Once installed, you can import Novu into your project and initialize it using your Novu account credentials. This step establishes a connection between your app and the Novu notification service.
+
+```python
+from novu.config import NovuConfig
+
+NovuConfig().configure("https://api.novu.co", "<YOUR_NOVU_API_KEY>")
+```
+
+Replace the `<YOUR_NOVU_API_KEY>` value with the authentic key from the **API Key** section of your [Novu Dashboard](https://web.novu.co/settings).
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:43]: Replace the `<YOUR_NOVU_API_KEY>` value with your own API key that you got from the **API Key** section of your [Novu Dashboard](https://web.novu.co/settings).

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+<aside>
+🔑 Note: Please do not hardcode your credentials in a file in production. Use environment variables instead.
+
+</aside>
+
+## Set Up A Channel Provider
+
+A channel provider is a service that provides one or more notification functionality such as sending an email, SMS, push notification etc. Our [integration store](https://web.novu.co/integrations) includes four channels: Email, SMS, Chat, and Push. These channels have multiple providers associated with them.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:52]: A channel provider is a service that lets you use one or more notification channels such as sending an email, SMS, push notification etc.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+| Channel | Providers                                                           |
+| ------- | ------------------------------------------------------------------- |
+| Email   | MailGun, Mandrill, MailJet, Amazon SES, Sendgrid, Postmark, Netcore |
+| SMS     | Twilio, Amazon SNS, Plivo, SMS, SMSCentral, Kannel, Infobip, Termii |
+| Chat    | Mattermost, Slack, Microsoft Teams, Discord                         |
+| Push    | FCM, APNS, Expo                                                     |
+
+Only one provider can be **active** per **channel**. Connect any of your favorite providers to get started. The email channel comes with Novu's email provider, which is active by default and includes 300 credits.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:61]: Only one provider can be **active** per **channel** at a time. Connect anyone of your preferred providers to get started. The email channel comes with Novu's email provider, which is active by default and includes 300 credits.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+## Create a Notification Workflow
+
+A notification workflow is the blueprint for the notifications that will be sent. It holds the entire flow of messages sent to the subscriber. This is where all the different channels are tied together under a single entity.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:65]: What is the last line supposed to mean here?

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+
+| Channel | Content Style                                                                                 |
+| ------- | --------------------------------------------------------------------------------------------- |
+| Email   | 1. Custom Code (HTML) with option to use custom variables via the handlebars , {{ }}, syntax. |
+|         | 2. Click and place UI items with the visual template editor.                                  |
+| SMS     | Text with the option to use handlebars syntax, {{ }} to inject custom variables.              |
+| Chat    | Text with the option to use handlebars syntax, {{ }} to inject custom variables.              |
+| In-App  | Text                                                                                          |
+
+Note: Proper authorization needs to be set for the Chat channel for subscribers.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:80]: > Note: Proper authorization needs to be setup for the Chat channel.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+Please proceed to create a notification workflow.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:82]: To create a notification workflow, please follow the following steps:

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+
+1. Click “Workflows” on the left sidebar of your Novu dashboard.
+2. Click the “Create Workflow” button on the top right.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:85]: After step 2, the third step is to select 'blank workflow' from the dropdown. Please update this

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+3. The name of a new workflow is currently "Untitled." Rename it to a more suitable title.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:86]: Rename it to a suitable title.

Comment (sumitsaurabh927) [reply docs/docs/overview/quickstart/get-started-with-python.md:86]: Also, images for the first three steps will be welcome! :)

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+4. Select "Email" as the channel you want to add.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:87]: Select any channel you want to use in your app. For the sake of this guide, we'll be using the 'Email' channel.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+   ![set-email.png] (https://res.cloudinary.com/dxc6bnman/image/upload/v1686776583/set-email_wavtrn.png)
+
+5. Click on the recently added channel, fill the email subject and click “Update”.
+   ![update_email_template.png](https://res.cloudinary.com/dxc6bnman/image/upload/v1686776583/update_email_template_ivn0jv.png)
+
+6. Click on the “Test” tab and send a test email to verify your notification workflow.
+   ![send_test_email.png](https://res.cloudinary.com/dxc6bnman/image/upload/v1686776583/send_test_email_ngzmth.png)
+
+You should get an email within seconds. Yaaay, you have successfully sent your first notification via the Novu dashboard! Now, let’s take it a step further to trigger notifications via code.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:96]: You should get an email within seconds. If you didn't, please check your 'spam' folder as sometimes test emails can end up there. Yaaay, you have successfully sent your first notification via the Novu dashboard! 

Now, let’s take it a step further to trigger notifications via code.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+
+## Create A Subscriber
+
+The recipients of a triggered notification are called subscribers.
+
+Click “Subscribers” on the left sidebar of the Novu dashboard to see all subscribers. By default, the dashboard will display a subscriber, as you were added automatically during sign-up.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:102]: Click “Subscribers” on the left sidebar of the Novu dashboard to see all subscribers. By default, the dashboard will display only one subscriber, as you were added automatically during sign-up.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+
+Now, let's create a subscriber on Novu. Copy and paste the following code to do so:
+
+```python
+# Create a subscriber
+
+from novu.api.subscriber import SubscriberApi
+from novu.dto.subscriber import SubscriberDto
+   
+your_subscriber_id = "123"; # Replace this with a unique user ID that matches your database.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:114]: @unicodeveloper I like the approach of @ankur0904 to extract the `subscriberID` into a separate variable (as it allows us to comment and tell the reader about it). Shall we update all the other quick start guides to use this very approach or do you feel the approach that we've been using is better and Ankur should change it to that style as well (for uniformity across guides)?

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+subscriber.create(SubscriberDto(
+        subscriber_id=your_subscriber_id,
+        email="abc@gmail.com",
+        first_name="John",
+        last_name="Doe"
+    )
+)
+```
+
+Run the code in your terminal like so:
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:126]: Run this code in your.....

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 18]
```diff
@@ -0,0 +1,198 @@
...
+
+```bash
+python main.py # replace main.py with your file name
+```
+
+You should see the subscriber on your Novu dashboard.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:132]: You should see a new subscriber (that you created above) on your Novu dashboard.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+
+## Update A Subscriber
+
+To update the Subscriber details you can follow you can call the `put` method from `SubcriberApi`. Here is an example:
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:136]: To update the Subscriber details you can call the `put` method from `SubcriberApi`. Here is an example:

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+from novu.api.subscriber import SubscriberApi
+from novu.dto.subscriber import SubscriberDto
+subscriber = SubscriberApi("https://api.novu.co", "<YOUR_NOVU_API_KEY>")
+subscriber.put(subscriber=SubscriberDto(
+        subscriber_id="<You want to update>",
+        email="abc@gmail.com",
+        first_name="John",
+        last_name="Doe"
+))
+```
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:150]: I think here, you aren't updating the subscriber as the `email`, `first_name` and `last_name` using the same values as when you had created the subscriber. Also the `subscriber_id` should be same as the one you're trying to update (which in this case is the `subscriber_id` from above.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 21]
```diff
@@ -0,0 +1,198 @@
...
+
+Other valid fields that can be updated are `email`, `first_name` etc.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:152]: Remove this line please

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 22]
```diff
@@ -0,0 +1,198 @@
...
+
+<aside>
+Note: To make all of your app users subscribers, you need to programmatically add them to Novu.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:155]: Note: To send notifications to all your users, you'll need to make them subscribers in Novu, which you can do by programmatically adding them to Novu.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 23]
```diff
@@ -0,0 +1,198 @@
...
+
+Run the code to trigger a notification!
+
+```bash
+python main.py # replace main.py with your file name
+```
+
+## Next Steps
+
+Great job! If you've reached this point, you should now have successfully Set up a channel provider, Create a notification workflow, Create a Subscriber, Update a Subscriber, Trigger a Notification in your application.
```
Comment (sumitsaurabh927) [inline docs/docs/overview/quickstart/get-started-with-python.md:193]: Great job! If you've reached this point, you should now have successfully Set up a channel provider, Created a notification workflow, Created a Subscriber, Updatde a Subscriber, Triggered a Notification in your application.

File: docs/docs/overview/quickstart/get-started-with-python.md (hunk 3 above)
Comment (ankur0904) [reply docs/docs/overview/quickstart/get-started-with-python.md:17]: @sumitsaurabh927 No, currently I don't made any demo repo. I'll made and update this.

File: docs/docs/overview/quickstart/get-started-with-python.md [hunk 24]
```diff
@@ -0,0 +1,198 @@
...
+
+Note: Proper authorization needs to be set for the Chat channel for subscribers.
+
+Please proceed to create a notification workflow.
+
+1. Click “Workflows” on the left sidebar of your Novu dashboard.
+2. Click the “Create Workflow” button on the top right.
+3. The name of a new workflow is currently "Untitled." Rename it to a more suitable title.
+4. Select "Email" as the channel you want to add.
+   ![set-email.png] (https://res.cloudinary.com/dxc6bnman/image/upload/v1686776583/set-email_wavtrn.png)
```
[the remark this answers is not in the document]
Comment (ankur0904) [reply docs/docs/overview/quickstart/get-started-with-python.md:88]: @sumitsaurabh927 I take the images from the **NodeJS** Quickstart. If this need to be changed or required any update. Please let me know.

File: docs/docs/overview/quickstart/get-started-with-python.md (hunk 16 above)
Comment (ankur0904) [reply docs/docs/overview/quickstart/get-started-with-python.md:114]: @sumitsaurabh927 Thanks

File: docs/docs/overview/quickstart/get-started-with-python.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (ankur0904) [reply docs/docs/overview/quickstart/get-started-with-python.md:91]: @sumitsaurabh927 One reason is when user switch to different quickstart guides they will feel consistent and it is also easy for browser to render the images by cache.

**Comment by ankur0904** [comment]
@sumitsaurabh927 Make changes as suggested

File: docs/docs/overview/quickstart/get-started-with-python.md (hunk 16 above)
Comment (unicodeveloper) [reply docs/docs/overview/quickstart/get-started-with-python.md]: its fine, other guides can still remain the way it is.

**Comment by sumitsaurabh927** [comment]
Hey @ankur0904 there are a few things you need to do:

1. I see you've used an actual API key in your guide. Please remove it and update the guide.
2. Every guide also needs a corresponding app. Please make an app (you'll own the repo for now and when this guide is merged, you'll transfer it to us). We had discussed it here: https://github.com/novuhq/novu/pull/3766/files#r1260675571

Happy to answer any more questions if you have! 🙂

**Comment by ankur0904** [comment]
@

> Replace the `<YOUR_NOVU_API_KEY>` value with your own API key that you got from the **API Key** section of your [Novu Dashboard](https://web.novu.co/settings).

@sumitsaurabh927 
1. Does I need to remove the actual API key with  `<YOUR_NOVU_API_KEY>`  ?
2. I have created the app [link](https://github.com/ankur0904/python-quickstart)

**Comment by sumitsaurabh927** [comment]
Yes. Use an env file 

Where's the link to the repo of app?

**Comment by sumitsaurabh927** [comment]
> https://github.com/ankur0904/python-quickstart

In the demo app repo, please add a README that illustrates how to run that app locally. You can use [this](https://github.com/novuhq/nextjs-quickstart) as a starting point. 🙂

**Comment by sumitsaurabh927** [comment]
> > https://github.com/ankur0904/python-quickstart
> 
> In the demo app repo, please add a README that illustrates how to run that app locally. You can use [this](https://github.com/novuhq/nextjs-quickstart) as a starting point. 🙂

Also, in the demo app, remove any sensitive data (like API key, subscriberID etc) and use dummy data instead and specify that one needs to use their own API key or subscriberID in the README.

**Comment by ankur0904** [comment]
@sumitsaurabh927 Okay

**Comment by ankur0904** [comment]
@sumitsaurabh927 Updated the  `README.md` and `main.py` in https://github.com/ankur0904/python-quickstart.

**Review by sumitsaurabh927 — APPROVED** [review]

**Comment by unicodeveloper** [comment]
Hey @ankur0904 So, I had to quickly create a new pr based off your work [here](https://github.com/novuhq/novu/pull/3899)

So, please can you look at the new PR and update it with the codesample?

Thank you

**Comment by ankur0904** [comment]
@unicodeveloper Yes, everything looks well in the new [PR](https://github.com/novuhq/novu/pull/3899) and it is merged. Does anything else I have to do?

**Comment by scopsy** [comment]
@unicodeveloper what do you want do with this one with the new docs migration?

**Comment by scopsy** [comment]
Closed as stale 🙏

**Pull request closed without merging**